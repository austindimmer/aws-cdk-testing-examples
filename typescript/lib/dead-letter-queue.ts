import * as cdk from "@aws-cdk/core";
import * as cloudwatch from "@aws-cdk/aws-cloudwatch";
import * as sqs from "@aws-cdk/aws-sqs";

export class DeadLetterQueue extends sqs.Queue {
  public readonly messagesInQueueAlarm: cloudwatch.IAlarm;

  constructor(scope: cdk.Construct, id: string) {
    super(scope, id);

    this.messagesInQueueAlarm = new cloudwatch.Alarm(this, "Alarm", {
      alarmDescription: "There are messages in the Dead Letter Queue.",
      evaluationPeriods: 1,
      threshold: 1,
      metric: this.metricApproximateNumberOfMessagesVisible(),
    });
  }
}
