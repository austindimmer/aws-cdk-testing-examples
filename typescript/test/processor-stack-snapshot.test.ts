import { Template } from "@aws-cdk/assertions";
import * as cdk from "@aws-cdk/core";
import * as sns from "@aws-cdk/aws-sns";
import { ProcessorStack } from "../lib/processor-stack";

describe("ProcessorStack", () => {
  test("matches the snapshot", () => {
    // Set up the app and resources in the other stack.
    const app = new cdk.App();
    const topicsStack = new cdk.Stack(app, "TopicsStack");
    const topics = [new sns.Topic(topicsStack, "Topic1", {})];

    // Create the ProcessorStack.
    const processorStack = new ProcessorStack(app, "ProcessorStack", {
      topics: topics, // Cross-stack reference
    });

    // Prepare the stack for assertions.
    const template = Template.fromStack(processorStack);

    // Assert the template matches the snapshot.
    expect(template.toJSON()).toMatchSnapshot();
  });
});
