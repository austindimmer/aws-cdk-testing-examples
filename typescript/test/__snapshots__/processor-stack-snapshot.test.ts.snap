// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessorStack matches the snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586ArtifactHash97E3F10B": Object {
      "Description": "Artifact hash for asset \\"6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586\\"",
      "Type": "String",
    },
    "AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3BucketE40C5F0B": Object {
      "Description": "S3 bucket for asset \\"6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586\\"",
      "Type": "String",
    },
    "AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3VersionKey189FBF34": Object {
      "Description": "S3 key for asset version \\"6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "LambdaFunctionAllowInvokeTopicsStackTopic1A54934186CE533EE": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "LambdaFunctionBF21E41F",
            "Arn",
          ],
        },
        "Principal": "sns.amazonaws.com",
        "SourceArn": Object {
          "Fn::ImportValue": "TopicsStack:ExportsOutputRefTopic198E71B3E74722020",
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "LambdaFunctionBF21E41F": Object {
      "DependsOn": Array [
        "LambdaFunctionServiceRoleDefaultPolicy32EEEE35",
        "LambdaFunctionServiceRoleC555A460",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3BucketE40C5F0B",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3VersionKey189FBF34",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3VersionKey189FBF34",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "STATE_MACHINE_ARN": Object {
              "Ref": "StateMachine2E01A3A5",
            },
          },
        },
        "Handler": "handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LambdaFunctionServiceRoleC555A460",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaFunctionServiceRoleC555A460": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaFunctionServiceRoleDefaultPolicy32EEEE35": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "states:StartExecution",
              "Effect": "Allow",
              "Resource": Object {
                "Ref": "StateMachine2E01A3A5",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaFunctionServiceRoleDefaultPolicy32EEEE35",
        "Roles": Array [
          Object {
            "Ref": "LambdaFunctionServiceRoleC555A460",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "LambdaFunctionTopic1461E184C": Object {
      "Properties": Object {
        "Endpoint": Object {
          "Fn::GetAtt": Array [
            "LambdaFunctionBF21E41F",
            "Arn",
          ],
        },
        "Protocol": "lambda",
        "TopicArn": Object {
          "Fn::ImportValue": "TopicsStack:ExportsOutputRefTopic198E71B3E74722020",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "StateMachine2E01A3A5": Object {
      "DependsOn": Array [
        "StateMachineRoleB840431D",
      ],
      "Properties": Object {
        "DefinitionString": "{\\"StartAt\\":\\"StartState\\",\\"States\\":{\\"StartState\\":{\\"Type\\":\\"Pass\\",\\"End\\":true}}}",
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "StateMachineRoleB840431D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::StepFunctions::StateMachine",
    },
    "StateMachineRoleB840431D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "states.",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ".amazonaws.com",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
